### DISCUSSION TITLE
Using Automerge in React Native

### DISCUSSION THREAD

**Discussion by momics** [automerge/automerge#893] [Q&A] [answered]
Hi there,

Today I was playing around a bit with automerge and I love it!

I would like to use it inside a React Native app. My thought was to use [this React Native Webassembly library](https://github.com/cawfree/react-native-webassembly) and then somehow make it so that @automerge/automerge-wasm uses that package to run the WASM backend.

I'm still a bit lost as to how I could get automerge to use a custom implementation of the (low level?) JS wrapper lib. I am unable to find where the source code for the JS side of the automerge-wasm package is located. I am talking about the source code that produces these JS files:

<img width="330" alt="image" src="https://github.com/automerge/automerge/assets/12343163/b9e0769a-28a9-47c2-97c8-0f5021109f04">

I could successfully load the wasm module on its own inside the React Native package, but I still lack the knowledge to actually bind the API correctly to JS.

Does anyone know where the source code for these JS files is located? Perhaps taking a look at the source for these modules I can better understand how to proceed.

Any other ideas on how I can get this working in RN ?

**Comment by alexjg** [comment, marked as answer]
The files you are looking at are autogenerated by [`wasm-bindgen`](https://github.com/rustwasm/wasm-bindgen), this means it's not straightforward to edit them. However, I think if you have managed to laod the wasm as a `WebAssembly.Module` then you should be able to bind that to the `automerge-wasm` package by importing the relevant JS file (e.g. `web/automerge_wasm.js`) and calling `initSync(<webassembly module>)`.

I do want to build first party support for this workflow. There are many places where it would be useful.

**Comment by hmbrg** [comment]
@Momics Did you make any progress on this? I'm also very interested in using automerge with react-native!
I was thinking more about an approach where we could use the rust library directly with expo-modules. There is this example repo (https://github.com/dgca/expo-rust-demo) that looks very interesing!

**Reply by momics** [reply]
I was able to get the web assembly module running in RN using the web assembly library above. However, I don't have sufficient knowledge of the low level API's to use it properly. I ran into some compatibility issues when trying to use the low level api with the rest of the automerge ecosystem (Repo / JS lib). Haven't really made any progress since then.
